File: Core/DTOs/BookBorrowRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace Core.DTOs
{
    public class BookBorrowRequestDto
    {
        [Required]
        public int BookId { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }
    }

    public class BorrowResultDto: ResultDto
    {
        public int BookId { get; set; }
        public string BookTitle { get; set; } = string.Empty;
        public string BookAuthors { get; set; } = string.Empty;
        public string MemberEmail { get; set; } = string.Empty;
        public string MemberFullName { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set;}
    }
}

File: client/src/app/shared/models/book-borrow-request-dto.ts
import { ResultDto } from "./result-dto";

export interface BookBorrowRequestDto {
    bookId: number;
    email: string;
    startDate: Date;
    endDate: Date;
};
  
export interface  BorrowResultDto extends ResultDto {
    bookId: number;
    bookTitle: string;
    bookAuthors: string;
    memberEmail: string;
    memberFullName: string;
    startDate: Date;
    endDate: Date;
};